workspaces/ballerina/ballerina-extension/src/features/ai/service/datamapper/context_api.ts

```diff
@@ -505,21 +381,21 @@ async function textExtractionUsingClaude({
 }
 
 // Utility functions for specific use cases
-export async function generateMappingInstruction(input: { file?: FileData; text?: string }): Promise<DataMapperResponse> {
+export async function generateMappingInstruction(input: { files?: FileData[]; text?: string }): Promise<DataMapperResponse> {
```

Comment:
cant we make files mandatory, we can pass empty array if itsn ot there. 
Check all casesl ike this